### DISCUSSION TITLE
Candlesticks custom PriceScale based on visible data

### DISCUSSION THREAD

**Discussion by christose** [tradingview/lightweight-charts#679] [General] [answered]
Hello,
I need candlesticks to be thin and tall and I understand that in order to achieve that, it is related to the visible range of the data and the price scale range. However, I've set autoScale: true and the following code:

```
this.ChartSeries.applyOptions({
      autoscaleInfoProvider: (original) => {
        var res = original();
        var visible = this.GetHighLow(this.VisibleData);
        if (res.priceRange !== null && visible != null) {
            res.priceRange.minValue = visible.Low;
            res.priceRange.maxValue = visible.High;
        }
        res.margins = {
          above: 0,
          below: 0
        };
        console.log(res.priceRange);
        return res;
      }
    });
```

But it doesn't seem to work. I realized the following, even if I get in the console the {minValue: 1.206, maxValue: 1.21666} which are correct based on the visible range, the price scale has the min and max value based on all the data that exist inside the chart, the data that are loaded but not visible. The case I am describing is related to the screenshot below.
image
![image](https://user-images.githubusercontent.com/8969889/105157584-cd325200-5b15-11eb-8f17-cc2ad3349318.png)

**Comment by timocov** [comment]
>  I've set autoScale: false and the following code:

If you'd like to use `autoscaleInfoProvider` you don't need to disable autoscale, because disabling it you disable calling your function (`autoscaleInfoProvider` allows you _override_ autoscale algorithm, but it should be enabled in the first place).

> the data that are loaded but not visible

Can you create a repro for that please? JSfiddle is enough.

**Comment by christose** [comment]
Sorry I meant to write true not false!
So autoScale is true and autoscaleInfoProvider function is being triggered normally where I return the high and low of the visible data but the priceScale instead shows on the price scale the low and high based on all the loaded data in the chart. 

Well it would be a pleasure to create a JSfiddle for you but its going to be time consuming to reproduce it there, to prepare the two days data and the code in typescript where one class extends the other.

Can I give you aceess on the platform so you can take a look on the actual code ? Can we try this way first ?
But I would need somewhere private a chat or email to give you url and user pass.

**Reply by timocov** [reply]
> Can I give you aceess on the platform so you can take a look on the actual code ? Can we try this way first ?

You can, but I think it will be much better to have independent from your platform reproduction (and we can use it in tests if it's a bug). Also, I don't think that you have there development build with non-minified code, but debugging well-minified lightweight-charts' code is quite complicated because all identifiers and properties are renamed...

**Reply by christose** [reply]
I have the development build with non-minified code on if that helps :)

I'm not very familiar with the JSfiddle doing something from scratch and I'm just trying to save me some time but if you dont like it that way I will go with the JSfiddle

**Reply by timocov** [reply]
Okay, let's try. You can send me the credentials/urls with access to `etimokhov@tradingview.com`.

**Reply by christose** [reply]
I have just sent the email, thank you!

**Comment by timocov** [comment, marked as answer]
After looking into the issue, the reason is having `rightPriceScale.scaleMargins`  options, which affects "free space" while autoscale is enabled. Not setting it or setting to `{ top: 0, bottom: 0 }` will fix the issue.

**Reply by christose** [reply]
Sorry I haven't posted the answer on Friday, I had a hardware issue with the pc which got just fixed now.
